server/src/main/java/org/elasticsearch/common/settings/ClusterSettings.java

```diff
@@ -514,9 +514,9 @@ public void apply(Settings value, Settings current, Settings previous) {
         IndexingPressure.MAX_INDEXING_BYTES,
         ShardLimitValidator.SETTING_CLUSTER_MAX_SHARDS_PER_NODE_FROZEN,
         DataTier.ENFORCE_DEFAULT_TIER_PREFERENCE_SETTING,
-        StableMasterHealthIndicatorService.IDENTITY_CHANGES_THRESHOLD_SETTING,
-        StableMasterHealthIndicatorService.NO_MASTER_TRANSITIONS_THRESHOLD_SETTING,
-        StableMasterHealthIndicatorService.NODE_HAS_MASTER_LOOKUP_TIMEFRAME_SETTING,
+        StableMasterService.IDENTITY_CHANGES_THRESHOLD_SETTING,
+        StableMasterService.NO_MASTER_TRANSITIONS_THRESHOLD_SETTING,
+        StableMasterService.NODE_HAS_MASTER_LOOKUP_TIMEFRAME_SETTING,
```

Comment:
Should we document these settings?

Reply:
I think we ought to. I've been trying to figure out where would be a good place to document these without cluttering more important documentation. Have any ideas? Maybe a settings page for the whole health API?

Reply:
Shall we have a page under https://www.elastic.co/guide/en/elasticsearch/reference/current/settings.html that's called something around what this service will end up being called? 
ie. cluster diagnostics settings? 

These settings will need some big warnings like [here](https://www.elastic.co/guide/en/elasticsearch/reference/current/modules-discovery-settings.html#_expert_settings)